### PR TITLE
[T2-Automation]: CEPH-83575816 - Radosgw bucket list should not be tr…

### PR DESCRIPTION
**Pull request by ckulal**
…uncated to 1000 entries

CEPH-83575816 - Radosgw bucket list should not be truncated to 1000 entries
Modify user with max_bucket quota set to 5000
and create 5000 buckets for a user.

Ensure with default value rgw_list_buckets_max_chunk:1000, all the 5000 buckets are listing via radosgw-admin bucket list

Pass log: http://magna002.ceph.redhat.com/ceph-qe-logs/Chaithra/CEPH-83575816/test_bucket_list.console.log

other old config logs:
http://magna002.ceph.redhat.com/ceph-qe-logs/Chaithra/CEPH-83575816/
failure not related to current changes

### CODE REVIEW

File: rgw/v2/tests/s3_swift/configs/test_bucket_list_not_truncated.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#script - test_bucket_listing.py
+# Polarion ID : CEPH-83575816
+config:
+   user_count: 1
+   bucket_count: 5000
+   objects_count: 0
```
Comment (viduship) [inline rgw/v2/tests/s3_swift/configs/test_bucket_list_not_truncated.yaml:6]: can we have at least 1 object per bucket here of 4kb size

File: rgw/v2/tests/s3_swift/test_bucket_listing.py [hunk 2]
```diff
@@ -197,88 +204,106 @@ def test_exec(config, ssh_con):
...
+                            cmd = "ceph daemon `ls -t /var/run/ceph/ceph-client.rgw.*.asok|head -1` config show |grep  rgw_bucket_index_max_aio"
+                            _, stdout, _ = ssh_con.exec_command(cmd)
+                            max_aio_output = stdout.readline()
+                        else:
+                            cmd = "ceph daemon `ls -t /var/run/ceph/ceph-client.rgw.*.asok|head -1` config show |grep  rgw_bucket_index_max_aio"
+                            max_aio_output = utils.exec_shell_cmd(cmd)
+                        max_aio = max_aio_output.split()[1]
                     else:
-                        cmd = "ceph daemon `ls -t /var/run/ceph/ceph-client.rgw.*.asok|head -1` config show |grep  rgw_bucket_index_max_aio"
+                        cmd = "ceph config get mon rgw_bucket_index_max_aio"
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/test_bucket_listing.py:229]: @ckulal , grep of rgw_bucket_index_max_aio from asok file was added for backward compatibility(nautilus), please retain the changes.

Comment (ckulal) [reply rgw/v2/tests/s3_swift/test_bucket_listing.py:229]: This checks with asok file is not removed, it is still available in line no: 225

**Comment by mkasturi18** [comment]
Changes looks good to me.

File: rgw/v2/tests/s3_swift/test_bucket_listing.py [hunk 3]
```diff
@@ -78,6 +78,13 @@ def test_exec(config, ssh_con):
             rgw_conn = auth.do_auth()
         objects_created_list = []
         bucket_created = []
+
+        if config.test_ops.get("modify_user", None) is True:
+            cmd = f"radosgw-admin user modify --uid={each_user['user_id']} --max-buckets {config.bucket_count}"
+            log.info("command is {cmd}")
+            modified_user = utils.exec_shell_cmd(cmd)
+            log.info(f"op is {modified_user}")
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/test_bucket_listing.py:86]: Can we have better log.info msg which is more understandable?

**Review by anrao19 — APPROVED** [review]

**Review by hmaheswa — APPROVED** [review]
LGTM

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**